draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,520 @@
+            "InterestGroup1"
+          ]
+        },
+        {
+          "tags": [
+            "keys"
+          ],
+          "data": [
+            "keyAfromInterestGroup1",
+            "keyBfromInterestGroup1"
+          ]
+        }
+      ]
+    },
+    {
+      "id": 1,
+      "compressionGroupId": 0,
+      "arguments": [
+        {
+          "tags": [
+            "interestGroupNames"
+          ],
+          "data": [
+            "InterestGroup2",
+            "InterestGroup3"
+          ]
+        },
+        {
+          "tags": [
+            "keys"
+          ],
+          "data": [
+            "keyMfromInterestGroup2",
+            "keyNfromInterestGroup3"
+          ]
+        }
+      ]
+    }
+  ]
+}
+~~~~~
+
+## Core Response Data {#response}
+
+The response is compressed. The compression is applied independently to each compression group. That
+means, the response object mainly contains a list of compressed blobs, each for one compression
+group. Each blob is for outputs of one or more partitions, sharing the same `compressionGroup` value
+as specified in the request.
+
+### Encryption {#response-encryption}
+
+The response uses the a similar encapsulated response format to that used by [OHTTP] (see
+{{Section 4.4 of OHTTP}}), but with the custom `message/ad-auction-trusted-signals-response`
+media type instead of `message/bhttp response`
+
+### Response Schema {#response-schema}
+
+The response is a [CBOR] encoded message with the following [CDDL] schema:
+
+~~~~~ cddl
+response = {
+  ? compressionGroups : [* compressionGroup]
+}
+
+compressionGroup = {
+  ? compressionGroupId: uint,
+  ; Partition outputs with the same `compressionGroupId` specified in the request
+  ; are compressed together.
+  ? ttl_ms: uint,
+  ; Adtech-specified TTL for client-side caching. In milliseconds. Unset means no caching.
+  ? content: bstrs
+  ; Compressed CBOR binary string using the algorithm specified in the request
+  ; For details see compressed response content schema below.
+}
+~~~~~
+
+##### CompressionGroup {#compression-group}
+
+The content of each `compressionGroup` is a serialized [CBOR] list of partition outputs. This object
+contains actual key value results for partitions in the corresponding compression group. The
+uncompressed, deserialized [CBOR] content has the following [CDDL] schema:
+
+~~~~~ cddl
+[* partitionOutput]
+; Array of PartitionOutput objects
+
+partitionOutput = {
+  ? id: uint
+  ; Unique id of the partition from the request
+  ? keyGroupOutputs: [* keyGroupOutput]
+}
+
+keyGroupOutput = {
+  tags: [* tstr]
+  ; List of tags describing this key group's attributes
+  ? keyValues: {
+    ; At least one key-value pair if present
+    * tstr => keyValue
+  }
+  ; One value to be returned in response for one key
+  ; If a keyValues object exists, it must at least contain one key-value pair. If no key-value pair can be returned, the key group should not be in the response
+}
+
+keyValue = {
+  value: tstr
```

Comment:
This is more of a question about the response format than the spec itself. Why do we have a dictionary containing a single string field instead of just putting the string field directly? It make be good to have a comment about this here in case someone asks.

Reply:
@peiwenhu I don't remember exactly why we had it like this. Do you?